packages/playwright-core/src/utils/network.ts

```diff
@@ -71,7 +71,7 @@ export function httpRequest(params: HTTPRequestParams, onResponse: (r: http.Inco
   const requestCallback = (res: http.IncomingMessage) => {
     const statusCode = res.statusCode || 0;
     if (statusCode >= 300 && statusCode < 400 && res.headers.location)
-      httpRequest({ ...params, url: res.headers.location }, onResponse, onError);
+      httpRequest({ ...params, url: new URL.URL(res.headers.location, params.url).toString() }, onResponse, onError);
```

Comment:
```suggestion
      httpRequest({ ...params, url: new URL(res.headers.location, params.url).toString() }, onResponse, onError);
```

change this to that and `import * as URL from 'url';` to `import * as url from 'url';`.

Clarified offline with Pavel that we want to continue to follow redirects.

Reply:
actually its fine, we used to do this before. Merging in.